featuretools/tests/primitive_tests/test_max_consecutive.py

```diff
@@ -11,230 +12,211 @@
-        array = pd.Series([1.0, -1.0, -2.0, 0.0, 5.0])
+        array = pd.Series([1.0, -1.0, -2.0, 0.0, 5.0], dtype=dtype)
         assert primitive_func(array) == 2
 
-    def test_with_nan(self):
-        primitive_instance = self.primitive()
+    def test_with_nan(self, dtype):
+        if dtype == "int64":
+            pytest.skip("nans not supported in int64")
+        primitive_instance = MaxConsecutiveNegatives()
         primitive_func = primitive_instance.get_function()
-        array = pd.Series([1, np.nan, -2, -3])
+        array = pd.Series([1, np.nan, -2, -3], dtype=dtype)
         assert primitive_func(array) == 2
 
-    def test_with_nan_skipna(self):
-        primitive_instance = self.primitive(skipna=False)
+    def test_with_nan_skipna(self, dtype):
+        if dtype == "int64":
+            pytest.skip("nans not supported in int64")
+        primitive_instance = MaxConsecutiveNegatives(skipna=False)
         primitive_func = primitive_instance.get_function()
-        array = pd.Series([-1, np.nan, -2, -3])
+        array = pd.Series([-1, np.nan, -2, -3], dtype=dtype)
         assert primitive_func(array) == 2
 
-    def test_all_nan(self):
-        primitive_instance = self.primitive()
+    def test_all_nan(self, dtype):
+        if dtype == "int64":
+            pytest.skip("nans not supported in int64")
+        primitive_instance = MaxConsecutiveNegatives()
         primitive_func = primitive_instance.get_function()
-        array = pd.Series([np.nan, np.nan, np.nan, np.nan])
+        array = pd.Series([np.nan, np.nan, np.nan, np.nan], dtype=dtype)
         assert np.isnan(primitive_func(array))
 
-    def test_all_nan_skipna(self):
-        primitive_instance = self.primitive(skipna=True)
+    def test_all_nan_skipna(self, dtype):
+        if dtype == "int64":
+            pytest.skip("nans not supported in int64")
+        primitive_instance = MaxConsecutiveNegatives(skipna=True)
         primitive_func = primitive_instance.get_function()
-        array = pd.Series([np.nan, np.nan, np.nan, np.nan])
+        array = pd.Series([np.nan, np.nan, np.nan, np.nan], dtype=dtype)
         assert np.isnan(primitive_func(array))
 
 
+@pytest.mark.parametrize("dtype", ["float64", "int64"])
 class TestMaxConsecutivePositives:
-    primitive = MaxConsecutivePositives
-
-    def test_regular(self):
-        primitive_instance = self.primitive()
-        primitive_func = primitive_instance.get_function()
-        array = pd.Series([1.3, -3.4, 1, 4, 10, -1.7, -4.9])
-        assert primitive_func(array) == 3
-
-    def test_all_int(self):
-        primitive_instance = self.primitive()
-        primitive_func = primitive_instance.get_function()
-        array = pd.Series([1, -1, 2, 4, -5])
-        assert primitive_func(array) == 2
-
-    def test_all_float(self):
-        primitive_instance = self.primitive()
+    def test_regular(self, dtype):
+        if dtype == "int64":
+            pytest.skip("floats not supported in int64")
```

Comment:
floats -> nans

Reply:
No this was intentional message. I made a better one
https://github.com/alteryx/featuretools/pull/2420/commits/a014517e40ce6dde0475b6c242bfe4566b183556